### DISCUSSION TITLE
ODF from grains meanOrientation?

### DISCUSSION THREAD

**Discussion by Jesusmunozmontecinos** [mtex-toolbox/mtex#1000] [Ask Anything]
Hello all,

I came here because I have a question (maybe misleading): Is it correct to plot an ODF (and a further PF) from the grains mean orientation?. My idea is to avoid the effect of larger grains (That can have a "higher" weight on the vissualization of CPO)

As for example: 

```
odf2_j = calcDensity(grains('jadeite').meanOrientation,'halfwidth',10*degree);
PFs_j_hkl = [ ...
Miller(1,0,0,ebsd('j').CS,'hkl'), ...
Miller(0,1,0,ebsd('j').CS,'hkl'), ...
Miller(0,0,1,ebsd('j').CS,'hkl'), ...
figure (3)
plotPDF(odf2_j,PFs_j_hkl,'antipodal','MarkerSize',1)
colorbar
annotate([xvector,yvector,zvector],...
'label',{'X','Z','Y'},'backgroundcolor','w','MarkerSize',20,'FontSize',20)
mtexColorbar
drawNow(gcm,'figSize','large')
hold off
```

Thanks and have a nice week :)

**Comment by lfgmorales** [comment]
Hi Jesus


Assuming that you calculate grains with a reasonable misorientation threshold (let's say, 10° or something like that), your script seems to be correct. However, don't do that if you are calculating any sort of physical property



best



Luiz


Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________
From: Jesusmunozmontecinos ***@***.***>
Sent: 03 May 2021 11:16:42
To: mtex-toolbox/mtex
Cc: Subscribed
Subject: [mtex-toolbox/mtex] ODF from grains meanOrientation? (#1000)


Hello all,

I came here because I have a question (maybe misleading): Is it correct to plot an ODF (and a further PF) from the grains mean orientation?. My idea is to avoid the effect of larger grains (That can have a "higher" weight on the vissualization of CPO)

As for example:

odf2_j = calcDensity(grains('jadeite').meanOrientation,'halfwidth',10*degree);
PFs_j_hkl = [ ...
Miller(1,0,0,ebsd('j').CS,'hkl'), ...
Miller(0,1,0,ebsd('j').CS,'hkl'), ...
Miller(0,0,1,ebsd('j').CS,'hkl'), ...
figure (3)
plotPDF(odf2_j,PFs_j_hkl,'antipodal','MarkerSize',1)
colorbar
annotate([xvector,yvector,zvector],...
'label',{'X','Z','Y'},'backgroundcolor','w','MarkerSize',20,'FontSize',20)
mtexColorbar
drawNow(gcm,'figSize','large')
hold off


Thanks and have a nice week :)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1000>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7XX34I7HN5HRZ7DRLTLZSXVANCNFSM44AO4DXQ>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1000", "url": "https://github.com/mtex-toolbox/mtex/discussions/1000", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Jesusmunozmontecinos** [reply]
Dear Luiz,

Thanks a lot for your answer,

hopping you a good week

Jesùs

**Comment by ralfHielscher** [comment]
Hi Jesus,

I not sure whether such an "ODF" computed from grain mean orientations makes physically sense. In the end the ODF should represent the distribution of lattice orientations within your specimen.

Ralf.

**Comment by Jesusmunozmontecinos** [comment]
Dear Ralf,

Thanks for your answer. What do you think is the best option in this case to avoid the effect of larger grains (and also to get a nicely contoured plot from continuous data such as does ODF?)

-The option that comes to my mind is to created a condition and exclude grains larger than X diameter or area; then make an ODF **without grains meanOrientation**

-A second option is for instance: **plotPDF(grains**('Glaucophane').**meanOrientation**,h,'MarkerSize',1); since it gives no weight to larger grains. But the data is not continuous as does the ODF

Probably I'm missing something conceptual, can you please help me?

**Reply by ralfHielscher** [reply]
I'm believe this requires some bootstrapping method. I'm not an expert in this.

Ralf.

**Reply by kilir** [reply]
Hi Jesus,
to me it sound like you are trying to deal with some sort of typical sampling artifact; you measure an limited area (because of time, available sample size etc.) and you have a microstructure with e.g. many many small grains and few large grains. If you believe both should belong together because they undergo the same processes or so, then obviously their size isn't so important, if you only have e.g. 3 large ones, you can't even figure out if both populations belong together because  that is just a very small numer and you just use the data from everything but the large grains - and still compute the ODF from those using each point.

In any cases, you can always compute an ODF, no matter if you use the orientations of a (subset) of your ebsd data or grain meanorientations or a subset of those.
```
ebsd_subset = ebsd(grains(grains.grainSize>xy))
% plot a map to see if this makes sense

% compute and odf
odf = calcDensity(ebsd_subset('phaseABC').orientations,'angle', smart_halfwidth_selection)
```

However, if it's not so simple, it may help if you can show an example microstructure from which you'd like to obtain your CPO information.

Cheers,
Rüdiger

**Comment by lfgmorales** [comment]
Just to complement what Rüdiger and Ralf said, we did a test using boostrapping 200 orientations 1000x from I think around 3000 grains we detected in a map, and we plotted the texture index, and you will see in the figure 4 of the attached paper that by calculating the mean orientation of grains, the texture index plot in the middle of the bootstrapping method. In this case we had enough grains, I tend to agree with Ralf here that physically that may not make sense, as an approximation, and now added what Rüdiger proposed, you may try to improve your results



http://www.gm.univ-montp2.fr/PERSO/mainprice/W_data/My_Publications_PDF/Morales_etal2018%20Olivine-antigorite%20orientation%20relationships.pdf


cheers



Luiz




Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by Jesusmunozmontecinos** [comment]
Dear Ralf, Luiz and Rüdiger,

Thanks you all for these very good and interesting questions,

I have solved the problem applying conditions and using a "normal" ODF without the grains meanOrientation, as pointed out by Rüdiger, the problem was a classical issue of different grain populations.

Although the problem was rather simple and my personal objective was trying to figure out the influence of the larger grains using the meanOrientation, I have seen a more technical issue:

In a monoclinic mineral such a omphacite (this example) I got the following PF:

![image](https://user-images.githubusercontent.com/76631957/118235768-7b7f5000-b495-11eb-8ab3-ce800e8b5ad6.png)

In the case of an ODF from orientations (TOP) the patterns are similar to what has been reported previously and in agreement with the studied context. However, the ODF from meanOrientation (bottom; the same patter is seen ploting meanOrientation without an ODF), shows [100] and [010] as sub parallels, this pattern has never been reported in omphacite, and it seems an artifact or a "mistake".

I'm wondering wether this pattern is really an effect of grain populations (and therefore my samples) since in the example from Luiz's paper, the patterns are consistent in monoclinic antigorite.

Thanks a lot for your time

**Reply by kilir** [reply]
Looking at the difference, I'd say you have a few (1,2) large grains.
Cheers,
Rüdiger
